### PR TITLE
MD-5: Principal evaluation

### PR DESCRIPTION
**Pull request by alexandre-merle**
Allow evaluate principal field of policies for iam roles

See https://scality.atlassian.net/browse/MD-5

### CODE REVIEW

File: lib/policyEvaluator/principal.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+     * @param {object} params - Evaluation parameters
+     * @param {object} statement - Statement policy field
+     * @param {object} valids - Valid principal fields
+     * @return {string} result of principal evaluation, either 'Neutral',
+     *  'Allow' or 'Deny'
+     */
+    static _evaluatePrincipalField(params, statement, valids) {
+        const reverse = !!statement.NotPrincipal;
+        const principal = statement.Principal || statement.NotPrincipal;
+        if (typeof principal === 'string' && principal === '*') {
```
Comment (ThibaultRiviere) [inline lib/policyEvaluator/principal.js]: I think the second condition is enough ^^

**Comment by alexandre-merle** [comment]
@rahulreddy i think you are the maintainer if i remember well :) can you have a look ? :)

**Comment by rahulreddy** [comment]
@alexandre-merle Yes I am maintaining the project. I am reviewing the code right now and will post the review soon.

File: lib/policyEvaluator/principal.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
+/**
+ * Class with methods to manage the policy 'principal' validation
+ */
+class EvaluatePrincipal {
```
[the remark this answers is not in the document]
Comment (alexandre-merle) [reply lib/policyEvaluator/principal.js]: done :)

**Comment by alexandre-merle** [comment]
@rahulreddy i will update the comments in the second PR, since arguments are changed :) if its okay for you

**Comment by rahulreddy** [comment]
@ironman-machine r+

**Comment by rahulreddy** [comment]
@ironman-machine r- r+

**Pull request closed without merging**